### PR TITLE
Fixes issue #4848 - Integrate REST module

### PR DESCRIPTION
**Pull request by manorrock**
Delivers an REST lifecycle for Faces

### CODE REVIEW

File: rest/src/main/resources/META-INF/beans.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<beans
+    xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/beans_2_0.xsd"
+    bean-discovery-mode="all">
+</beans>
```
Comment (BalusC) [inline rest/src/main/resources/META-INF/beans.xml:8]: Please use CDI 3.0 one.

File: rest/src/main/resources/META-INF/faces-config.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version='1.0' encoding='UTF-8'?>
+
+<faces-config version="2.3"
+              xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+              xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-facesconfig_2_3.xsd">
+</faces-config>
```
Comment (BalusC) [inline rest/src/main/resources/META-INF/faces-config.xml:7]: Please use JSF 3.0 one.

Comment (manorrock) [reply rest/src/main/resources/META-INF/faces-config.xml:7]: Done

File: rest/src/main/resources/META-INF/beans.xml (hunk 1 above)
Comment (manorrock) [reply rest/src/main/resources/META-INF/beans.xml:8]: Done

Comment (BalusC) [reply rest/src/main/resources/META-INF/beans.xml:8]: Looks like you only adjusted the one in `/cdi`. Please also adjust abovementioned one :)

File: rest/src/main/resources/META-INF/faces-config.xml (hunk 2 above)
Comment (BalusC) [reply rest/src/main/resources/META-INF/faces-config.xml:7]: Looks like you only adjusted the one in `/cdi`. Please also adjust abovementioned one :)

File: rest/src/main/resources/META-INF/beans.xml (hunk 1 above)
Comment (manorrock) [reply rest/src/main/resources/META-INF/beans.xml:9]: Done

File: rest/src/main/resources/META-INF/faces-config.xml (hunk 2 above)
Comment (manorrock) [reply rest/src/main/resources/META-INF/faces-config.xml:8]: Done

**Pull request merged**